addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rotelra1x.handler;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.rotelra1x.internal.ConfigurationError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.PortInUseException;
+import gnu.io.RXTXPort;
+import gnu.io.SerialPort;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * The {@link RotelRa1xHandler} handles commands and state changes, and communicates
+ * with the amplifier over the serial port interface..
+ *
+ * @author Marius Bjørnstad - Initial contribution
+ */
+public class RotelRa1xHandler extends BaseThingHandler implements Runnable {
+
+    private static final int BAUD = 115200;
+    private static final long ERROR_RETRY_DELAY_MS = 60000;
+    private int maximumVolume;
+    private RXTXPort serialPort;
+
+    private boolean exit;
+    private volatile boolean power;
+
+    private Logger logger = LoggerFactory.getLogger(RotelRa1xHandler.class);
+
+    public RotelRa1xHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        maximumVolume = ((BigDecimal) getThing().getConfiguration().get("maximum-volume")).intValue();
+        exit = false;
+        // Seems we need to wait a bit after initialization for the channels to
+        // be ready to accept updates, so deferring input loop by 4 sec.
+        scheduler.schedule(this, 4, TimeUnit.SECONDS);
```

Comment:
It should not be scheduled on a configuration error for sure. The only way to fix a config involves calling initialize() again. If the cable happens to be disconnected on start-up, for a communication error, this means that there's no way to recover. I've changed it anyway, since that's quite unlikely to happen.